src/environmentd/src/http/webhook.rs

```diff
@@ -170,7 +170,9 @@ impl From<StorageError> for WebhookError {
     fn from(err: StorageError) -> Self {
         match err {
             // TODO(parkmycar): Maybe map this to a HTTP 410 Gone instead of 404?
-            StorageError::IdentifierMissing(id) => WebhookError::NotFound(id.to_string()),
+            StorageError::IdentifierMissing(id) | StorageError::IdentifierInvalid(id) => {
```

Comment:
CC (@MaterializeInc/storage) is it intentional that both of these errors exist? The descriptions make it seem like they are the same thing

> /// The source identifier is not present.

and 

> /// The provided identifier was invalid, maybe missing, wrong type, not registered, etc.

Reply:
I actually created this error in a previous PR based on feedback. The two errors are quite similar, but they do convey slightly different things, not opposed to getting rid of `IdentifierInvalid` if folks think they're too similar.

Theoretically we could return a `400` for `IdentifierInvalid` instead of a `404`, but at a different point in webhook handling we already check to make sure the source is the correct type and if not return a `400` there. So if we're receiving `StorageError::IdentifierInvalid` it means the source has gone missing.